File: 70. Climbing Stairs/solution.py
class Solution:
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        prev1 = 1
        prev2 = 2
        if n == 1:
            return prev1
        
        if n == 2:
            return prev2
            
        for i in range(3, n+1):
            tmp = prev1 + prev2
            prev1 = prev2
            prev2 = tmp
            
        return prev2
            
        

File: 70. Climbing Stairs/solution.java
class Solution {
    public int climbStairs(int n) {
        int step1 = 1;
        if (n == 1) {
            return step1;
        }
        
        int step2 = 2;
        if (n == 2) {
            return step2;
        }
        
        n -= 2;
        
        while (n > 0) {
            step2 = step2 + step1;
            step1 = step2 - step1;
            n--;
        }
        
        return step2;
    }
}
